ZuluContent/Mobiles/PlayerMobile/PlayerMobile.cs

```diff
@@ -1220,8 +1215,33 @@ public override void OnDeath(Container c)
                 if (master != null)
                     killer = master;
             }
-        }
 
+            if (FreeDeath)
```

Comment:
This could probably be a buff with an `OnDeath` hook. 

Also if you haven't already tried it, it might work better inside of `Kill()` as you can prevent the corpse being made and having to transfer the items back to the player as the ZH93 version doesn't make corpse.

Reply:
Problem is Kill is on Mobile so it would be quite a mission to add this in there.

Atm logic is the same as ZH, where it still shows the death effect momentarily and just moves all the items from corpse and deletes the corpse

Reply:
Sorry I meant overriding `Kill()` as it's virtual, you would conditionally call the base method.

Reply:
But what about about the death sound and animation, logic to move items to the player's backpack, create a death robe, etc? Would have to rewrite this...

I also kind of like how you actually die first, then immediately are resurrected, if your stuff just goes to your pack it doesn't really feel like you died imo

Reply:
Sure I'm not really bothered either way was just curious if there was a cleaner way to implement it, let's stick with `OnDeath` then.

In that case do you want to change `bool willDie` property for the `OnDeath` hook to `bool resurrect` since they do technically die they just get immediately resurrected?

Reply:
yup can do